src/Microsoft.AspNet.SignalR.Client/Transports/LongPollingTransport.cs

```diff
@@ -239,16 +240,23 @@ private Task OnAfterPoll(Exception exception)
         // internal virtual for testing
         internal virtual void OnError(IConnection connection, Exception exception)
         {
-            TransportFailed(exception);
+            // Prevent the reconnecting -> connected transition from happening if it hasn't already.
             _reconnectInvoker.Invoke();
 
+            if (TryFailStart(exception))
+            {
+                return;
+            }
+
             if (!TransportHelper.VerifyLastActive(connection))
             {
                 StopPolling();
             }
-
-            // Transition into reconnecting state
-            connection.EnsureReconnecting();
+            else
```

Comment:
Why is this changing?

Reply:
When VerifyLastActive returns false it should move to the disconnected state which should cause EnsureReconnecting to no-op. I was just making it more explicit in an attempt to make this code slightly easier to reason about.

Reply:
Alright cool, I figured it was something like that, just checking